### PR TITLE
Backfill Clair V4 Manifest Security Information 

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Effectively moves completely to using Clair v4 for manifest security information. The `securityworker` will start indexing all manifests in the database with Clair v4, and Quay's API will return Clair v4 info if it has been indexed. Manifests that have previously been indexed with Clair v2 will continue to return Clair v2 info until it has been indexed with Clair v4. All non-read requests to Clair v2 will be ceased.

### Changes

- [x] Removes `SECURITY_SCANNER_V4_NAMESPACE_WHITELIST` (all namespaces indexed now)
- [x] `V2SecurityScanner.perform_indexing()` throws `NotImplementedError`
- [x] Removes use of `SplitScanToken` and thows `AssertionError` if you try and perform indexing for Clair v2
- [x] Removed `security_notification_worker` and associated code
- [x] Removed `register_model_cleanup_callbacks` from security scanner interface 
- [x] Removed deletion code from Clair v2 API client

Addresses https://issues.redhat.com/browse/PROJQUAY-625

### CODE REVIEW

File: data/secscan_model/__init__.py [hunk 1]
```diff
@@ -17,63 +18,53 @@
...
 
         return self
 
     def perform_indexing(self, next_token=None):
-        if next_token is None:
-            return SplitScanToken("v4", self._v4_model.perform_indexing(None))
-
-        if next_token.version == "v4" and next_token.token is not None:
-            return SplitScanToken("v4", self._v4_model.perform_indexing(next_token.token))
+        if next_token is None or next_token.version != "v4":
```
Comment (josephschorr) [inline data/secscan_model/__init__.py]: Can't we just send this to V4 directly now?

File: tox.ini [hunk 2]
```diff
@@ -21,7 +21,7 @@ setenv =
 commands =
     python --version
     alembic upgrade head
-    py.test --timeout=3600 --cov-report=html --cov-report=term-missing -x --color no --verbose {env:FILE} -vv {posargs}
+    /bin/sh -c "py.test --timeout=3600 --cov-report=html --cov-report=term-missing -x --color yes --verbose {env:FILE} -vv {posargs}"
```
Comment (josephschorr) [inline tox.ini]: Why this change?

Comment (alecmerdler) [reply tox.ini]: Whoops

File: config.py [hunk 3]
```diff
@@ -475,9 +474,6 @@ def create_transaction(db):
     # The version of the API to use for the security scanner.
     SECURITY_SCANNER_API_VERSION = "v1"
 
-    # Namespace whitelist for security scanner.
-    SECURITY_SCANNER_V4_NAMESPACE_WHITELIST = []
```
Comment (alecmerdler) [inline config.py:480]: Are we allowed to remove fields from `config.py`?

File: data/secscan_model/__init__.py (hunk 1 above)
Comment (alecmerdler) [reply data/secscan_model/__init__.py]: If we want to get rid of the `SplitToken` completely now, then yes.

**Review by alecmerdler — COMMENTED** [review]
I've deployed this change using a k8s Quay installation and confirmed its functionality.

File: data/secscan_model/__init__.py (hunk 1 above)
Comment (josephschorr) [reply data/secscan_model/__init__.py]: Yes please. I don't think we need it anymore.

**Comment by alecmerdler** [comment]
@josephschorr I'm fairly confident this is ready to go.

File: util/secscan/api.py [hunk 4]
```diff
@@ -74,27 +74,20 @@ def __init__(self, response):
 _API_METHOD_INSERT = "layers"
 _API_METHOD_GET_LAYER = "layers/%s"
 _API_METHOD_DELETE_LAYER = "layers/%s"
```
Comment (josephschorr) [inline util/secscan/api.py]: Remove the delete layer and all the deletion code as well. We should stop calling delete in Clair V2.

Comment (alecmerdler) [reply util/secscan/api.py]: Removed the `register_model_cleanup_callbacks` from the security scanner interface.

Comment (josephschorr) [reply util/secscan/api.py]: Please remove the delete layer call as well

Comment (alecmerdler) [reply util/secscan/api.py]: Removed all code from `ImplementedSecurityScannerAPI` that performs any sort of write operation on the API.

File: data/secscan_model/interface.py [hunk 5]
```diff
@@ -40,15 +40,6 @@ def perform_indexing(self, start_token=None):
         to any particular format.
         """
 
-    @abstractmethod
```
Comment (josephschorr) [inline data/secscan_model/interface.py]: We still need this for deletion for V4

Comment (alecmerdler) [reply data/secscan_model/interface.py]: Added back this method.

**Review by josephschorr — APPROVED** [review]
LGTM
